src/MCPClient/lib/clientScripts/archivematicaCreateMETSMetadataXML.py

```diff
@@ -0,0 +1,265 @@
+    objects/dir/file.pdf,file_metadata_b.xml,metadata_type_b
+
+    Example dict returned:
+
+    {
+        "objects": {"metadata_type": Path("/path/to/objects_metadata.xml")},
+        "objects/dir": {"metadata_type": Path("/path/to/dir_metadata.xml")},
+        "objects/dir/file.pdf": {
+            "metadata_type_a": Path("/path/to/file_metadata_a.xml"),
+            "metadata_type_b": Path("/path/to/file_metadata_b.xml"),
+        },
+    }
+
+    :param str sip_path: Absolute path to the SIP.
+    :param bool reingest: Boolean to indicate if it's a reingest.
+    :return dict, list: Dictionary with File/dir path -> dict of type -> metadata
+    file pathlib Path, and list with errors (if a CSV row is missing the filename
+    or type, or if there is more than one entry for the same filename and type).
+    """
+    mapping = {}
+    errors = []
+    source_metadata_paths = []
+    metadata_path = Path(sip_path) / "objects" / "metadata"
+    transfers_metadata_path = metadata_path / "transfers"
+    if reingest:
+        source_metadata_paths.append(metadata_path / "source-metadata.csv")
+    elif transfers_metadata_path.is_dir():
+        for dir_ in transfers_metadata_path.iterdir():
+            source_metadata_paths.append(dir_ / "source-metadata.csv")
+    for source_metadata_path in source_metadata_paths:
+        if not source_metadata_path.is_file():
+            continue
+        with source_metadata_path.open() as f:
+            reader = csv.DictReader(f)
+            for row in reader:
+                if not all(k in row and row[k] for k in ["filename", "type"]):
+                    errors.append(
+                        "A row in {} is missing the filename and/or type".format(
+                            source_metadata_path
+                        )
+                    )
+                    continue
+                if row["type"] == "CUSTOM":
+                    errors.append(
+                        "A row in {} is using CUSTOM, a reserved type".format(
+                            source_metadata_path
+                        )
+                    )
+                    continue
+                if row["filename"] not in mapping:
+                    mapping[row["filename"]] = {}
+                elif row["type"] in mapping[row["filename"]]:
+                    errors.append(
+                        "More than one entry in {} for path {} and type {}".format(
+                            source_metadata_path, row["filename"], row["type"]
+                        )
+                    )
+                    continue
+                if row["metadata"]:
+                    row["metadata"] = source_metadata_path.parent / row["metadata"]
+                mapping[row["filename"]][row["type"]] = row["metadata"]
+    return mapping, errors
+
+
+def _get_schema_uri(tree):
+    XSI = "http://www.w3.org/2001/XMLSchema-instance"
+    VALIDATION = mcpclient_settings.XML_VALIDATION
```

Comment:
Do you think is worth trying to avoid the `mcpclient_settings` import and cascade this as a parameter from the caller to the top level function and to here?